BookEcommerce/Controllers/VendorController.cs

```diff
@@ -0,0 +1,42 @@
+﻿using BookEcommerce.Models.DTOs;
+using BookEcommerce.Models.DTOs.Response.Base;
+using BookEcommerce.Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace BookEcommerce.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+
+    [Authorize(Roles = "VENDOR", AuthenticationSchemes = "Bearer")]
+    public class VendorController : ControllerBase
+    {
+        private readonly IVendorService vendorService;
+        public VendorController(IVendorService vendorService)
+        {
+            this.vendorService = vendorService;
+        }
+        [Authorize(AuthenticationSchemes = "Bearer", Roles = "VENDOR")]
+        [HttpPost("create")]
+        public async Task<IActionResult> CreateProfile([FromForm] CreateVendorDTO CreateVendorDTO)
```

Comment:
I don't think it is a good route. POST vendor but it is createProfile instead of createVendor.

Reply:
updated